src/str.js

```diff
@@ -1363,6 +1363,9 @@ Sk.builtin.str.prototype.nb$remainder = function (rhs) {
             return "%";
         }
     };
+    if (index < rhs.sq$length()) {
+        throw Sk.builtin.TypeError("not all arguments converted during string formattin");
+    }
```

Comment:
can we fix this message (`'formatting'`) and use `'new'`
Also make sure that this passes the closure compiler linting checks, which seems to have failed somewhere

Reply:
Oh, when tests use python2 grammar, has some error like:
```shell
FAILED: (test/run/t166.py)
-----
print "%x" % 10
print "%x" % 10L
print "%x" % 100000000000L
print "%o" % 10
print "%o" % 10L
print "%o" % 100000000000L
print "%d" % 10
print "%d" % 10L
print "%d" % 100000000000L

-----
GOT:
-----
EXCEPTION: TypeError: not all arguments converted during string formattin on line 1

-----
WANTED:
-----
a
a
174876e800
12
12
1351035564000
10
10
100000000000


```
I need to check it.